### DISCUSSION TITLE
How to use useMany hook with ids that may be duplicates?

### DISCUSSION THREAD

**Discussion by SadmanYasar** [refinedev/refine#6411] [Q&A]
Currently, if I have an array of ids e.g [52,52,52], the useMany hook only includes one object matching the id instead of three objects for three ids in the array. I am using Directus data provider. Is this related to Refine or Directus SDK? The code is pretty much the same as the [docs](https://refine.dev/docs/data/hooks/use-many/#usage)

**Comment by BatuhanW** [comment]
Hey @SadmanYasar can you explain a bit more? getMany filters by given ids, having duplicate ids could be a problem.

**Comment by identity-wael** [comment]
If the duplicates represent repeated references in your UI, fetch the records once and then reconstruct the display sequence from the original IDs.

`useMany` delegates to the provider's `getMany`; it doesn't promise one response element for every occurrence of an ID. In the community Directus provider, [`getMany` uses an `_in` filter](https://github.com/tspvivek/refine-directus/blob/6f8a882b7fea6d044c95d85abb098cb7cb01b676/package/src/dataProvider.ts#L240-L263). A membership filter for `[52, 52, 52]` matches the record with ID 52 once. This is different from calling `getOne(52)` three times. Refine's [provider delegation is here](https://github.com/refinedev/refine/blob/2352eb5b6539e2f39ad9aef652279ad1dcf2c467/packages/core/src/hooks/data/useMany.ts#L190-L208).

For example, after fetching the unique IDs:

```ts
const ids = [52, 52, 52];
const uniqueIds = [...new Set(ids)]; // pass this to useMany

// `records` is the array returned by the provider.
const byId = new Map(records.map(record => [String(record.id), record]));
const occurrences = ids.map((id, index) => ({
  occurrenceKey: `${id}:${index}`,
  record: byId.get(String(id)),
}));
```

That produces three occurrences in the requested order without three identical requests. Keep missing records explicit (for example, show an unavailable placeholder) rather than silently changing the positions. Use an occurrence key when rendering; `key={record.id}` would be duplicated. If these occurrences can be reordered/edited independently, use their own stable relationship/line-item IDs instead of an array index.

The result property is version-dependent: in Refine v4 it's typically `data?.data`; current v5 exposes `result.data`. The reconstruction above is independent of that API difference. If you're using a different Directus provider, check its `getMany` implementation, but the same approach works for a set-based response.
